threat_intel/virustotal.py

```diff
@@ -105,6 +203,48 @@ def get_ip_reports(self, ips):
 
         return all_responses
 
+    @MultiRequest.error_handling
+    def get_file_search(self, query):
+        """Performs advanced search on samples, matching certain binary/
+           metadata/detection criteria.
+           Possible queries: file size, file type, first or last submission to
+            VT, number of positives, bynary content, etc.
+
+        Args:
+            query: dictionary with search arguments
+            Example: 'query': 'type:peexe size:90kb+ positives:5+ behaviour:"taskkill"'
+        Returns:
+            A dict with the VT report.
+        """
+        api_name = 'virustotal-file-search'
+        (all_responses, query) = self._bulk_cache_lookup(api_name, query)
+
+        response_chunks = self._request_reports("query", query, 'file/search')
+
+        self._extract_response_chunks(all_responses, response_chunks, api_name)
+
+        return all_responses
+
+    @MultiRequest.error_handling
```

Comment:
Similarly here the both methods follow the same pattern. That has a potential for a refactoring, but I wouldn't prioritize on that.